src/client/reducers/audioReducer.js

```diff
@@ -0,0 +1,67 @@
+import { CREATE_AUDIO_DATA, PLAY_PAUSE_SOUND_FROM_FILE, CREATE_STREAME_DATA, START_MUTE_STREAME_AUDIO, MERGE_CANVAS_WIDTH } from '../actions/types';
+
+const context = new (window.AudioContext || window.webkitAudioContext)();    
+const analyser = context.createAnalyser();
+
+const initialState = {  
+    //graphic canvas 
+     widthCanvas: 500,
+     heightCanvas: 150, 
+    //audio from file
+     trackName: undefined,     
+     trackType: undefined,
+     trackSize: undefined,     
+     audioContext: context,
+     analyser: analyser,
+     audioFile: undefined,
```

Comment:
If you all of all props with undefined is actually undefined, why do you need them in unit state?
You can ad them later

Reply:
I think that it is easier when anybody fo our team can see what props will be in store and can use them.

Reply:
its not good idea to init your state or with undefined or even use undefined, better to use will bu null or empty object

Reply:
ok